### PR TITLE
Start observing the Proxy object

### PR DESCRIPTION
**Pull request by stlaz**
Adds an observer for the Proxy object, pulls its fields and passes
them to the environment of the deployed openshift-apiserver container.

### CODE REVIEW

**Comment by stlaz** [comment]
Pulled in latest lib-go for the Proxy observer

File: pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go [hunk 1]
```diff
@@ -21,6 +22,7 @@ import (
 	apiregistrationv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
 	apiregistrationv1client "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/typed/apiregistration/v1"
 
+	"github.com/ghodss/yaml"
```
Comment (sttts) [inline pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go]: nit: order

File: pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go [hunk 2]
```diff
@@ -397,3 +409,17 @@ func resourceSelectorForCLI(obj runtime.Object) string {
 	}
 	return kind + group + "/" + name
 }
+
+func proxyMapToEnvVars(proxyConfig map[string]string) []corev1.EnvVar {
+	envVars := []corev1.EnvVar{}
```
Comment (sttts) [inline pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go]: move down where it is used, and return nil in the if-statement. Easier to read...

File: pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go [hunk 3]
```diff
@@ -326,6 +328,16 @@ func manageOpenShiftAPIServerDaemonSet_v311_00_to_latest(client appsclientv1.Dae
...
 
+	var observedConfig map[string]interface{}
+	if err := yaml.Unmarshal(operatorConfig.Spec.ObservedConfig.Raw, &observedConfig); err != nil {
+		return nil, false, fmt.Errorf("failed to unmarshal the observedConfig: %v", err)
+	}
+	proxyConfig, _, err := unstructured.NestedStringMap(observedConfig, "workloadcontroller", "proxy")
+	if err != nil {
+		return nil, false, fmt.Errorf("couldn't get the proxy config from observedConfig: %v", err)
+	}
+	required.Spec.Template.Spec.Containers[0].Env = append(required.Spec.Template.Spec.Containers[0].Env, proxyMapToEnvVars(proxyConfig)...)
```
Comment (sttts) [inline pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go]: should be append these to all containers? The cert syncher does not need it. But this kind of kind has bitten us before. I would prefer it to be generic.

Comment (stlaz) [reply pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go]: agree, that's probably the better thing to do

**Comment by stlaz** [comment]
addressed the comments

**Comment by stlaz** [comment]
/retest
servicemonitor race

**Comment by stlaz** [comment]
/retest
The servicemonitor race seems to appear quite often

**Comment by stlaz** [comment]
Hm, there's actually a bug in this PR, ProxyLister won't be callable.

**Pull request merged**